dasharo-compatibility/ec-and-super-IO.robot

```diff
@@ -272,3 +272,98 @@ ECR021.001 Not charging between 95% and 98% in OS (Ubuntu 22.04)
         Sonoff Power On
         Check Charging State Not Charging In Linux
     END
+
+ECR022.001 EC sync update with power adapter connected works correctly
+    [Documentation]    This test aims to verify whether coreboot update
+    ...    will also update EC firmware when power adapter is connected.
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    ECR022.001 not supported
+    Skip If    not ${DTS_FIRMWARE_FLASHING_SUPPORT}    ECR022.001 not supported
+    Skip If    not ${DTS_EC_FLASHING_SUPPORT}    ECR022.001 not supported
+
+    Mount Image On PiKVM    ${PIKVM_IP}    dts-v1.2.21.img
+    # Flash old fw version without ec sync
+    Make Sure That Flash Locks Are Disabled
+    Power On
+    Boot Dasharo Tools Suite    USB
```

Comment:
Can we boot iPXE over the network and copy the required coreboot_with_ec.rom file over scp instead? Right now it looks like the test requires a specially prepared DTS image

Reply:
@mkopec 
When I was testing iPXE didn't work but if it works now then yes we can.
I'm not sure if scp will work I couldn't manage to use it with robot keywords (or any other file transfer). Even using it manually to copy from host to DUT didn't work if I remember right it's because DTS prints welcome message on each ssh connection which breaks scp. The only way I could send files was to use scp on DUT to download file from host.
I guess we could wget this file from server if it's available.

Reply:
@pietrushnic I must've forgotten it. Issue is here along with fix/workaround: https://github.com/Dasharo/dasharo-issues/issues/840

Reply:
@mkopec changed in [e18f683](https://github.com/Dasharo/open-source-firmware-validation/pull/276/commits/e18f68315a38d11c02e9f863f3cb599e54a3fef8) but wasn't tested on platform, just partially on QEMU.
Also had to add changes in [9248bc1](https://github.com/Dasharo/open-source-firmware-validation/pull/276/commits/9248bc14a50ba6f29f1666c21d19ae214ef01b9f) because of this error:

```
Parent suite setup failed:
Resolving variable '${{$url.rsplit('/', 1)[1]}}' failed: Evaluating expression '$url.rsplit('/', 1)[1]' failed: SyntaxError: invalid syntax (<string>, line 1)
```

Reply:
Thanks